catalpainternational/canoe
# Teaching Activity completion persistance

Fixes #458 

This uses the `page.complete` mechanism to query and set completion of Teaching Activities.

When `page.complete` is set to true the mechanism built into all `Page` instances in previous PRs, stores the completion in idb, sends it to the api ( which will resync if offline when you come back online ) and sets redux state.
When you get `page.complete` you read from redux state.
On initial page load we read from the api into idb and from idb into redux state.

- completing all activities in a topic completes that topic ( and uncompleting one, uncompletes the topic )
- LongCard now supports the same behaviour as Card reducing prop passing ( progress bar capability is present but commented )

TEst:
- set a lesson activity to complete, refresh, it should still be complete
- set the same lesson activity to incomplete, refresh it should still be incomplete
- login, logout, the same completions should be set
- check the completion ticks on the topic page, and the teaching home comply with expectations

You can also check the data stored in your actions idb, and your UserAction server side database
